bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/LedgerDirsManager.java

```diff
@@ -290,6 +299,39 @@ public void addToWritableDirs(File dir, boolean underWarnThreshold) {
         }
     }
 
+    /**
+     * Add the dir to warn dirs list.
+     */
+    @VisibleForTesting
+    public void addToWarnDirs(File dir) {
+        if (!warnDirs.contains(dir)) {
+            LOG.warn(dir + " usages is above warn threshold. Adding it to warn dirs list");
+            List<File> updatedFilledDirs = new ArrayList<File>(warnDirs);
```

Comment:
Why not use `warnDirs.add(dir)` directly? Due to the warnDirs is a list, does it will happen the same dir being add to the list twice?

Reply:
Yes, it would be addToWarnDirs every time its usage exceeds the warn threshold